roles/delete_nodepools/tasks/main.yml

```diff
@@ -32,5 +34,4 @@
   loop: "{{ nodepool_zone1_id + nodepool_zone2_id }}"
   loop_control:
     loop_var: pool_id
-    when: not ansible_check_mode and k8s_cluster_id is defined and pool_id is defined
-
+    when: not ansible_check_mode and k8s_cluster_id is defined and pool_id is defined
```

Comment:
If all of the conditions specified in the "when" statement are not true, it will not  result in an error. this Ansible task  will simply be skipped and not executed.